### ISSUE TITLE
Add `GH_USERNAME` as default Vault value

### ISSUE THREAD

**Issue by abangser** [kubefirst/kubefirst#1710]
### What is your feature idea?

**tl;dr:**
Add the GitHub username as a value in Vault so that operators can use that if they need to interact with GitHub from inside the cluster.

**Implementation:**
When a cluster is created, there should be a GitHub username value somewhere in Vault, preferably near the GH Token value that is already stored.

I believe that Kubefirst already has this information, but even if you only have token, it is possible to generate the username from a curl request:
```
curl -Ls \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer ${GITHUB_TOKEN}" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/user | jq -r '.'
```

Thanks!

### Why is it needed?

Right now there are a number of useful default values in Vault. One of those values is the GitHub token used during cluster creation. This can be useful for cluster operators to interact with GitHub using other in cluster tools, however these interactions often require both the token and the username.

More specifically, I am a contributor to the tool Kratix which is in your catalog and requires users to provide GitHub credentials. My goal is to reduce user inputs by autogenerating the resources they need based on the Kubefirst credentials.

As you will see in the implementation details above, this is able to be generated by the operators, but feels nicer to not have to generate it given the other useful data is already in Vault.

### Is this missing feature preventing you from using kubefirst?

- [ ] Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fharper** [comment]
Thanks for the idea @abangser. As discussed on our Slack community, not sure when we'll be able to prioritize this, but I think it's a good feature idea.

**Comment by fharper** [comment]
It's not exactly what you wanted, but it is now accessible for applications installed from the [GitOps Catalog with the new tokens available](https://github.com/kubefirst/gitops-catalog#kubefirst-tokens). It may solve your issue if you needed that for Kratix deployment.
